frontend/src/components/modules/Tournaments/OngoingTournament/PlayoffTournamentView.tsx

```diff
@@ -1,31 +1,94 @@
-import TournamentBracket from "./TournamentBracket";
-import "./PlayoffTournamentView.css";
+import Bracket from "./TournamentBracket";
+import { type User, type Match } from "types/models";
+import { useTournament } from "context/TournamentContext";
+import { Typography, Box, Grid, Divider } from "@mui/material";
+
+interface Rounds extends Record<number, Match[]> {}
 
 const PlayoffTournamentView: React.FC = () => {
-  const initialPlayers = [
-    "Player 1",
-    "Player 2",
-    "Player 3",
-    "Player 4",
-    "Player 1",
-    "Player 2",
-    "Player 3",
-    "Player 4"
-  ];
-  const initialWinners: string[] = [];
+  const tournament = useTournament();
+  const matchSchedule: Match[] = tournament.matchSchedule;
+  const players: User[] = tournament.players;
+
+  // Calculate the total number of rounds, assuming it's a single-elimination tournament
+  const totalRounds = Math.ceil(Math.log2(players.length));
+
+  // Group matches by tournamentRound
+  const rounds: Rounds = matchSchedule.reduce<Rounds>((acc, match) => {
+    const round = match.tournamentRound;
+    if (acc[round] === undefined) {
+      acc[round] = [];
+    }
+    acc[round].push(match);
+    return acc;
+  }, {});
 
   return (
-    <div>
-      <TournamentBracket
-        players={initialPlayers}
-        initialWinners={initialWinners}
-      />
-    </div>
+    <Box
+      sx={{
+        overflowX: "auto",
+        "&::-webkit-scrollbar": { display: "none" }
+      }}
+    >
+      <Grid
+        container
+        spacing={2}
+        justifyContent="flex-start"
+        alignItems="flex-start"
+      >
+        {Object.entries(rounds).map(([roundNumber, matches], index) => (
+          <React.Fragment key={roundNumber}>
+            {index > 0 && <Divider orientation="vertical" flexItem />}
+            <Grid item>
+              <Box
+                sx={{
+                  display: "flex",
+                  flexDirection: "column",
+                  alignItems: "center",
+                  minWidth: 300
+                }}
+              >
+                <Typography
+                  variant="h6"
+                  sx={{
+                    marginBottom: 2,
+                    textAlign: "center",
+                    textDecoration: "underline"
+                  }}
+                >
+                  {parseInt(roundNumber) === totalRounds
+                    ? "Final"
+                    : `Round ${roundNumber}`}
+                </Typography>
+                {matches.map((match: Match) => {
+                  const tempPlayers: User[] = match.players.map(
+                    (matchPlayer) => {
+                      const player = players.find(
+                        (p) => p.id === matchPlayer.id
+                      );
+                      if (player === null || player === undefined) {
+                        throw new Error(
+                          `Player with ID ${matchPlayer.id} not found`
+                        );
```

Comment:
This will crash the entire app due to not being handled anywhere. I would instead show an error modal (you can use the one defined in the common components). You can find an example from the tournamentContext file for example.

Reply:
And regarding what to guide the user to do with the modal; I would either refresh the page, or offer the solution to go back (what I have done at least). We should try to limit any awkward situations where the user might get stuck or something.

Reply:
This will not occur. Because if there's a match between 2 players, then that player has to be in the players list. But lint kept throwing errors for a null checker. But I will add an error modal.

Reply:
If you are 100% sure, but keep in mind that this will crash everything

Reply:
I would rather show a modal that an unexpected error occurred and redirect to the home page.

Reply:
Although we can keep this like it is. Ideally we would have a last case scenario error handler, which would catch any errors and show the modal.